plugins/populator/frontend/public/javascripts/countly.views.js

```diff
@@ -22,6 +22,16 @@ window.PopulatorView = countlyView.extend({
 
 
         $(this.el).html(this.template(this.templateData));
+
+        if (countlyGlobal.apps[countlyCommon.ACTIVE_APP_ID].locked) {
+            $("#populator-locked").show();
+            $("#populator > .content").hide();
+        }
+        else {
+            $("#populator-locked").hide();
+            $("#populator > .content").show();
+        }
+
```

Comment:
Maybe additionally to browser side protection, we can also do server side protetions
1) Add populator=true parameter to all SDK populator requests
2) On server side check for each SDK request and if app is locked and request contains populator=true, then block request (both /i and /i/bulk) like this: https://github.com/Countly/countly-server/blob/master/api/utils/requestProcessor.js#L2034

That will also help us identify populator requests in Event logs too

Reply:
I wanted to block populator SDK requests server-side as well but I couldn't decide if it fit in the scope of this feature's requirements or was worth the overhead it would add to the SDK request processor. I'm down to implement it in this PR if you authorize it :)